lib/open_food_network/reports/line_items.rb

```diff
@@ -25,6 +23,16 @@ def self.list(permissions, params)
         end
         line_items
       end
+
+      def self.line_items_with_hidden_details(permissions, line_items)
+        editable_line_items = permissions.editable_line_items.pluck(:id)
+
+        if editable_line_items.empty?
+          line_items
+        else
+          line_items.where('"spree_line_items"."id" NOT IN (?)', editable_line_items)
```

Comment:
:warning: this will cause an N+1. We'll need to weight the benefits of leveraging the permissions memoization versus introducing these N queries to see if it's really worth shipping it as it is...

If I'm not mistaken the call to `Enumerable#select` on line 15 will execute the AR built up until then, which in turn will cause the AR above to be executed for every step of the iteration. We should be able to INNER JOIN `line_items` with `editable_line_items` before that and make PostgreSQL return only the data we care about. Do you see any problems on that?